src/date.ts

```diff
@@ -98,6 +98,28 @@ function fromString(date: string): Option<Date> {
     }
 }
 
+function formatSeconds(seconds: string): string | null {
+    try {
+        const secondsInt = parseInt(seconds);
+        if (isNaN(secondsInt)) {
+            return null;
+        }
+        const hrs = Math.floor((secondsInt / 3600));
+        const mins = Math.floor(((secondsInt % 3600) / 60));
+        const secs = Math.floor(secondsInt) % 60;
+        let ret = "";
+
+        if (hrs > 0) {
+            ret += `${hrs}:${mins < 10 ? "0" : ""}`
+        }
+
+        ret += `${mins}:${secs < 10 ? "0" : ""}${secs}`;
+        return ret;
+    } catch(e) {
+        return null;
+    }
+}
```

Comment:
Is it worth using our `Option` type here?

```suggestion
function formatSeconds(seconds: string): Option<string> {
    try {
        const secondsInt = parseInt(seconds);
        if (isNaN(secondsInt)) {
            return none;
        }
        const hrs = Math.floor((secondsInt / 3600));
        const mins = Math.floor(((secondsInt % 3600) / 60));
        const secs = Math.floor(secondsInt) % 60;
        let ret = "";

        if (hrs > 0) {
            ret += `${hrs}:${mins < 10 ? "0" : ""}`
        }

        ret += `${mins}:${secs < 10 ? "0" : ""}${secs}`;
        return some(ret);
    } catch(e) {
        return none;
    }
}
```